lib/state_of_the_nation.rb

```diff
@@ -30,6 +30,14 @@ def until(finish_key)
           (finish.blank? || round_if_should(finish) > round_if_should(time)) && round_if_should(start) <= round_if_should(time)
         end
 
+        define_method "active_in_interval?" do |interval_start:, interval_end: nil|
+          start_before_interval_finish_or_comparison_with_nil = (start.nil? || interval_end.nil? || round_if_should(start) < interval_end)
```

Comment:
This is real awkward to reason about 😢 

I didn't want it to become a crazy single logic line so tried to provide intermediate named variables. I'm unsure about whether that improved clarity though.